polars/polars-time/src/windows/groupby.rs

```diff
@@ -465,7 +461,11 @@ pub(crate) fn groupby_values_iter<'a>(
     // we have a (partial) lookbehind window
     if offset.negative {
         // only lookbehind
-        if offset.nanoseconds() == period.nanoseconds() {
+        if (matches!(closed_window, ClosedWindow::Right | ClosedWindow::Both)
+            && offset.nanoseconds() > period.nanoseconds())
+            || (matches!(closed_window, ClosedWindow::Left | ClosedWindow::None)
+                && offset.nanoseconds() >= period.nanoseconds())
+        {
```

Comment:
for a full lookbehind, the window must be behind the given point

if `offset` == `period`, and `closed` includes the right endpoint, then the window would include the given point, and so it wouldn't be a full lookbehind but just a partial one. If `closed` if `'left'`, on the other hand, then it's always a full lookbehind, so long as `offset >= period`